### ISSUE TITLE
[Elixir] Suggest new exercise: recursion

### ISSUE THREAD

**Issue by neenjaw** [exercism/v3#1490]
This issue describes how to implement the `numbers` concept exercise for the Elixir track.

[🔗 **How to help**](#actionable-items)

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise to learn how to do simple recursion in Elixir.

## Learning objectives

- Know how write a recursive function
- Know how to call a recursive function
- Know how recursion is often used to process lists

## Out of scope

- higher order functions to control recursion

## Concepts

- `recursion`

## Prerequisites

- `lists`
- `pattern-matching`

## Narrative for exercise

Following the ["Garden Bird Count"](https://github.com/exercism/v3/blob/master/reference/stories/arrays.garden-bird-count.md) narrative ([C# implmentation](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/arrays/.docs/instructions.md)), adapt this situation to recursing through lists.

## Resources to refer to

- <https://elixir-lang.org/getting-started/recursion.html>
- <https://elixirschool.com/blog/recursion/>
- <https://medium.com/@dlite/your-first-elixir-recursive-function-c2309d091bc6>
- <https://blog.appsignal.com/2019/03/19/elixir-alchemy-recursion.html>

### Hints

- slight hints to guide toward the solution without giving away exact functions/code/implmentation

### After

- recursion is how loops are performed in elixir
- recursion on enumerable data structures is common

## Representer

- TBD

## Analyzer

- check to see if Enum or List module functions used rather than coding intended recursive functions

## Actionable items <a href="#actionable-items" id="actionable-items">🔗</a>

looking for help in a few ways, and I am here to help you through getting this exercise off the ground:

- [ ] adapt the ["Garden Bird Count"](https://github.com/exercism/v3/blob/master/reference/stories/arrays.garden-bird-count.md) narrative ([C# implmentation](https://github.com/exercism/v3/blob/master/languages/csharp/exercises/concept/arrays/.docs/instructions.md)) to elixir
  - Write the exemplar solution (How can we fulfil the premise of the exercise with following the scope of requirements)
  - If we can't fulfil the premise with the current scope, then we need to adjust:
    - the premise
    - the code
    - the requirements
  - let's talk about it! We can help you!
- [ ] Writing the test cases (if TDD is your thing!)
- [ ] Writing the student facing documents
  - I am happy to do this for/with you!

For a first phase review we need:

- [ ] the [design.md document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-metadesignmd-file) (the information in this post)
- [ ] the exemplar solution
- [ ] the [introduction document](https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md#add-docsintroductionmd-file) describing the concepts to be taught

Thanks for considering!

## Need Help?

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by neenjaw** [comment]
This exercise could also introduce `charlists` as an extension to lists, as a premise for a recursive coding problem

**Comment by angelikatyborska** [comment]
@neenjaw Did you have any specific idea for how to add `charlists` into the Garden Bird Count story? Or would you rather that we come up with a different story?

**Comment by neenjaw** [comment]
I think i meant to type, charlists could  be a subsequent exercise because of the concept recursion being taught here.  Im not sure how charlists would work for this exercise.  My bad

**Comment by angelikatyborska** [comment]
Oh, ok. In that case, I will start working on this concept exercise using the Garden Bird Count story.
